pkg/terminal/proxy.go

```diff
@@ -0,0 +1,175 @@
+package terminal
+
+import (
+	"crypto/tls"
+	"errors"
+	"net/http"
+	"net/url"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/proxy"
+)
+
+const (
+	//Endpoint that Proxy is supposed to handle
+	Endpoint = "/api/terminal/"
+)
+
+// Proxy provides handlers to handle terminal related requests
+type Proxy struct {
+	TLSClientConfig *tls.Config
+	ClusterEndpoint *url.URL
+}
+
+var (
+	WorkspaceGroupVersionResource = schema.GroupVersionResource{
+		Group:    "workspace.che.eclipse.org",
+		Version:  "v1alpha1",
+		Resource: "workspaces",
+	}
+
+	UserGroupVersionResource = schema.GroupVersionResource{
+		Group:    "user.openshift.io",
+		Version:  "v1",
+		Resource: "users",
+	}
+)
+
+// Handle evaluates the namespace and workspace names from URL and after check that
+// it's created by the current user - proxies the request there
+func (p *Proxy) Handle(user *auth.User, w http.ResponseWriter, r *http.Request) {
```

Comment:
We need to restrict this to an HTTP method like POST that has CSRF checks

Reply:
Should be fixed now